### ISSUE TITLE
ZC1098 false positive: eval with command substitution

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1371]
ZC1098 ("Use `(q)` flag for quoting variables in eval") fires on `eval` arguments that contain a command substitution `$(...)`, where the suggested `(q)` flag does not apply.

## Reproduce

```zsh
eval "$(pyenv init - zsh)"
eval "$(rbenv init - zsh)"
eval "$(fzf --zsh)"
```

Each is flagged with "Use the `(q)` flag ... when using variables in `eval`". This is the standard shell-initialisation idiom emitted by pyenv, rbenv, nodenv, fzf, zinit, and zsh-vi-mode.

## Cause

`checkZC1098` scans the argument string for any `$` byte and treats a `$` that begins a command substitution `$(...)` the same as a parameter expansion `$name`.

## Why it is wrong

The `(q)` family of flags (`(q)`, `(qq)`, `(q-)`) are parameter-expansion flags: `${(q)var}`. They cannot be applied to a command substitution `$(...)`. The rule's own remedy is inapplicable, so flagging `$(...)` produces advice the reader cannot act on. Command substitution in `eval` is a distinct concern with a different remedy and is out of scope for this rule.

## Fix

Only set the variable flag for a `$` that introduces a parameter expansion, that is, a `$` not immediately followed by `(`. This preserves every genuine case: `eval "$var"`, `eval "${arr[$i]}"`, and mixed arguments such as `eval "$(cmd) ${userdata}"` (the `${userdata}` still fires).

## Related, narrower edge (not addressed here)

A single-quoted literal such as `eval 'print $HOME'` also fires. Single and double quotes share one token type with the quote character stripped, so the quoting context is not recoverable at the check layer. Distinguishing it requires carrying the quote kind on the string token and is tracked separately.
